classes/Tools.php

```diff
@@ -780,33 +780,6 @@ public static function convertPrice($price, $currency = null, $to_currency = tru
         return $price;
     }
 
-    /**
-     * Implement array_replace for PHP <= 5.2
-     *
-     * @return array|mixed|null
-     */
-    public static function array_replace()
```

Comment:
we can deprecate it but not remove it: think about all modules that may use it :+1:

Reply:
as you can see, we use it in core (but you remove this call in the next contribution) but we can't update every modules that comes from contributors.

Reply:
@mickaelandrieu Ok, thank you! Just for my information: do deprecated functions stay in the codebase forever? I saw Symfony for example removes deprecated code (on major versions?)